============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-8.3.3, pluggy-1.5.0
rootdir: /home/wilkinson/Documents/TCC/TG
collected 5 items

QuixBugs/python_testcases/test_pascal.py .FFFF                           [100%]

=================================== FAILURES ===================================
______________________ test_pascal[input_data1-expected1] ______________________

input_data = [2], expected = [[1], [1, 1]]

    @pytest.mark.parametrize("input_data,expected", testdata)
    def test_pascal(input_data, expected):
>       assert pascal(*input_data) == expected

QuixBugs/python_testcases/test_pascal.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

n = 2

    def pascal(n):
        rows = [[1]]
        for r in range(1, n):
            row = []
            for c in range(r + 1):
                upleft = rows[r - 1][c - 1] if c > 0 else 0
>               upright = rows[r - 1][c]
E               IndexError: list index out of range

QuixBugs/python_programs/pascal.py:8: IndexError
______________________ test_pascal[input_data2-expected2] ______________________

input_data = [3], expected = [[1], [1, 1], [1, 2, 1]]

    @pytest.mark.parametrize("input_data,expected", testdata)
    def test_pascal(input_data, expected):
>       assert pascal(*input_data) == expected

QuixBugs/python_testcases/test_pascal.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

n = 3

    def pascal(n):
        rows = [[1]]
        for r in range(1, n):
            row = []
            for c in range(r + 1):
                upleft = rows[r - 1][c - 1] if c > 0 else 0
>               upright = rows[r - 1][c]
E               IndexError: list index out of range

QuixBugs/python_programs/pascal.py:8: IndexError
______________________ test_pascal[input_data3-expected3] ______________________

input_data = [4], expected = [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1]]

    @pytest.mark.parametrize("input_data,expected", testdata)
    def test_pascal(input_data, expected):
>       assert pascal(*input_data) == expected

QuixBugs/python_testcases/test_pascal.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

n = 4

    def pascal(n):
        rows = [[1]]
        for r in range(1, n):
            row = []
            for c in range(r + 1):
                upleft = rows[r - 1][c - 1] if c > 0 else 0
>               upright = rows[r - 1][c]
E               IndexError: list index out of range

QuixBugs/python_programs/pascal.py:8: IndexError
______________________ test_pascal[input_data4-expected4] ______________________

input_data = [5]
expected = [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1], [1, 4, 6, 4, 1]]

    @pytest.mark.parametrize("input_data,expected", testdata)
    def test_pascal(input_data, expected):
>       assert pascal(*input_data) == expected

QuixBugs/python_testcases/test_pascal.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

n = 5

    def pascal(n):
        rows = [[1]]
        for r in range(1, n):
            row = []
            for c in range(r + 1):
                upleft = rows[r - 1][c - 1] if c > 0 else 0
>               upright = rows[r - 1][c]
E               IndexError: list index out of range

QuixBugs/python_programs/pascal.py:8: IndexError
=========================== short test summary info ============================
FAILED QuixBugs/python_testcases/test_pascal.py::test_pascal[input_data1-expected1]
FAILED QuixBugs/python_testcases/test_pascal.py::test_pascal[input_data2-expected2]
FAILED QuixBugs/python_testcases/test_pascal.py::test_pascal[input_data3-expected3]
FAILED QuixBugs/python_testcases/test_pascal.py::test_pascal[input_data4-expected4]
========================= 4 failed, 1 passed in 0.04s ==========================
Different AST


tenta resolver qualquer outra coisa
